src/services/imaged-object.ts

```diff
@@ -0,0 +1,141 @@
+
+    constructor(private store: Store<any>) {
+        this.communicator = new Communicator(store);
+    }
+
+    public async fetchImagedObjectInfo(editionId: number, imagedObjectId: string) {
+        let imagedObject = this._getCachedImagedObject(editionId, imagedObjectId);
+        if (!imagedObject) {
+            imagedObject = await this._getImagedObject(editionId, imagedObjectId);
+        }
+
+        if (!imagedObject) {
+            throw new NotFoundError('imagedObject', imagedObjectId);
+        }
+        const artefacts = await this.getImagedObjectArtefacts(editionId, imagedObject);
+        imagedObject.artefacts = artefacts;
+
+        this.store.dispatch('imagedObject/setImagedObject', imagedObject);
+
+        return imagedObject;
+    }
+
+    public async getImagedObjectArtefacts(editionId: number, imagedObject: ImagedObject): Promise<Artefact[]> {
+        if (!imagedObject.recto) {
+            console.error('ImagedObject ', imagedObject, ' has no recto information');
+            throw new Error('ImagedObject has no recto information');
+        }
+
+        const response = await CommHelper.get<ImagedObjectDTO>(
+            `/v1/editions/${editionId}/imaged-objects/${imagedObject.id}?optional=artefacts&optional=masks`
+        );
+
+        let artefacts: Artefact[] = [];
+        if (response.data.artefacts) {
+            artefacts = response.data.artefacts.map((obj: any) => new Artefact(obj)).filter((a) => a.side === 'recto');
+        }
+
+        return artefacts;
+    }
+
+    public async createArtefact(editionId: number, imagedObject: ImagedObject, artefactName: string):
+        Promise<Artefact> {
+        // const mask = artefact.mask ? artefact.mask.wkt : '';
+        let masterImageId = 0;
+        imagedObject.recto!.images.forEach((element) => {
+            if (element.master) {
+                masterImageId = element.id;
+            }
+        });
+        const body = {
+            masterImageId,
+            mask: '',
+            name: artefactName,
+            position: '1'
+
+        } as CreateArtefactDTO;
+        const response = await CommHelper.post<ArtefactDTO>(`/v1/editions/${editionId}/artefacts`, body);
+
+        const artefact = new Artefact(response.data);
+        return artefact;
+    }
+
+    public async changeArtefact(editionId: number, artefact: Artefact):
+        Promise<ArtefactDTO> {
+        const mask = artefact.mask ? artefact.mask.wkt : '';
+        const body = {
+            mask,
+            name: artefact.name,
+            position: '1', // TODO: what is position?
+        } as UpdateArtefactDTO;
+
+        const response = await CommHelper.put<ArtefactDTO>(`/v1/editions/${editionId}/artefacts/${artefact.id}`, body);
+        return response.data;
+    }
+
+    public async changeArtefactPosition(editionId: number, artefact: Artefact):
+        Promise<ArtefactPositionChangedResult> {
+        const transformMatrix = artefact.transformMatrix ?
```

Comment:
I don't want to get into matrix validations for now. The frontend will just pass empty strings around, and we'll worry about them later.

Reply:
Sure, once you have the matrix class going, it can handle all this.  The matrices will always be calculated by standard tooling, so I don't think we have to worry in the front-end about validating the matrices (but I will have to add that into the backend API).  
When would you pass this a null artefact.transform?  It seems this function should refuse to run unless a non-null artefact.transform is passed to it (but as you note, you don't need to verify the actual stringified matrix).